### ISSUE TITLE
[a11y]: [A][1.3.1, 4.1.2] Caption settings modal window is incorrectly implemented

### ISSUE THREAD

**Issue by MaksymZahreba** [videojs/video.js#9060]
### Description

Player noted to fail several WCAG checkpoints.

Actual Result: Selected options are announced as "Selected, Checked". Modal window implemented with incorrect tags and roles. "Reset" and "Done" options has incorrect announcement. 

![Image](https://github.com/user-attachments/assets/090c7854-1477-453a-b45e-ada991084c5b)

Expected Result: Selected option in dropdown should have aria-selected="true" attribute and should be announced as "Selected". Modal window should be implemented as it described [https://www.w3.org/WAI/ARIA/apg/patterns/dialog-modal/examples/dialog/. ](https://www.w3.org/WAI/ARIA/apg/patterns/dialog-modal/examples/dialog/)"Reset" should be announced just as "Reset button" and "Done" as "Done button".

Recommendations: 

Button that opens dialog should have aria-haspopup="dialog".

Dialog window widget should be implemented using WAI-ARIA pattern - https://www.w3.org/TR/wai-aria-practices-1.1/examples/dialog-modal/dialog.html

- The element that serves as the dialog container should have a role of [dialog](https://www.w3.org/TR/wai-aria-1.1/#dialog).
- All elements required to operate the dialog should be descendants of the element that has role dialog.
- The dialog container element should have [aria-modal](https://www.w3.org/TR/wai-aria-1.1/#aria-modal) set to true.
- The dialog should have either:
       - A value set for the [aria-labelledby](https://www.w3.org/TR/wai-aria-1.1/#aria-labelledby) property that refers to a visible dialog title.
      -  A label specified by [aria-label](https://www.w3.org/TR/wai-aria-1.1/#aria-label).

When the dialog is opened focus should be set on the "x" ("Close dialog") button.

Main heading of the dialog should be implemented as heading level 2 and should have ID attribute that is references in aria-labelledby attribute of the element that serves as the dialog container.

When a dialog closes, focus returns to the element that invoked the dialog unless the invoking element no longer exists. Then, focus is set on another element that provides logical work flow.

Failed WCAG Checkpoint(s):

1.3.1 Info and Relationships (Level A) https://www.w3.org/TR/WCAG22/#info-and-relationships

4.1.2 Name, Role, Value (Level A) https://www.w3.org/TR/WCAG22/#name-role-value

### Reduced test case

https://videojs.com/advanced/?video=bipbop-advanced

### Steps to reproduce

1. Open the page https://videojs.com/advanced/?video=bipbop-advanced using Win 11 + Chrome + JAWS
2. Using Tab key navigate to "Play" button to playback the video
3. Navigate to "CC" dropdown and note announcement
4. Navigate to "Caption settings" option and activate to open modal window

### What version of Video.js are you using?

latest

### What browser(s) including version(s) does this occur with?

Chrome 137

### What OS(es) and version(s) does this occur with?

Win 11

**Comment by MaksymZahreba** [comment]
Hi, any progress on the issue?
